src/browser/auto-updater-linux.coffee

```diff
@@ -0,0 +1,115 @@
+{EventEmitter} = require 'events'
+_ = require 'underscore-plus'
+ChildProcess = require 'child_process'
+app = require 'app'
+
+distros = {
+  ubuntu: /(\S*ubuntu)/i,
+  debian: /(debian)/i,
+  redhat: /(red(\s?)hat)/i,
+  centos: /(centos)/i,
+  fedora: /(fedora)/i,
+}
+
+spawn = (command, args, callback) ->
+  stdout = ''
+
+  try
+    spawnedProcess = ChildProcess.spawn(command, args)
+  catch error
+    # Spawn can throw an error
+    process.nextTick -> callback?(error, stdout)
+    return
+
+  spawnedProcess.stdout.on 'data', (data) ->
+    stdout += data
+
+  error = null
+  spawnedProcess.on 'error', (processError) -> error ?= processError
+  spawnedProcess.on 'close', (code, signal) ->
+    error ?= new Error("Command failed: #{signal ? code}") if code isnt 0
+    error?.code ?= code
+    error?.stdout ?= stdout
+    callback?(error, stdout)
+
+class AutoUpdater
+  _.extend @prototype, EventEmitter.prototype
+
+  stdout = ChildProcess.execSync('cat /etc/issue').toString().replace(/\\./g, "")
+
+  if distros.ubuntu.test(stdout) or distros.debian.test(stdout)
+    @pkg = 'atom-amd64.deb'
+    @pkgType = 'deb'
+    @installCmd = 'dpkg --install -y'
+  else if distros.redhat.test(stdout) or distros.centos.test(stdout)
+    @pkg = 'atom.x86_64.rpm'
+    @pkgType = 'rpm'
+    @installCmd = 'yum localinstall -y'
+  else if distros.fedora.test(stdout)
+    @pkg = 'atom.x86_64.rpm'
+    @pkgType = 'rpm'
+    @installCmd = 'dnf install -y'
+  else distro = 'unsupported'
+
+  setFeedUrl: (@updateUrl) ->
+
+  quitAndInstall: ->
+    app.once 'will-quit', -> spawn 'sudo', [installCmd, @pkg, '&&',
+      'rm', '-rf', @pkg, '&&', 'atom']
+    app.quit()
+
+  downloadUpdate: (callback) ->
+    spawn 'curl', ['-L', @updateUrl], (error, stdout) ->
```

Comment:
Is there any chance at all that `curl` isn't installed?  If so we'll need to add it as a dependency.